src/mqttcm_connect.c

```diff
@@ -1082,10 +1124,20 @@ rbusError_t MqttSubscribeSetHandler(rbusHandle_t handle, rbusProperty_t prop, rb
 					}
 					subscribe = strdup(data);
 					free(data);
+					if(clientId !=NULL)
+					{
+						snprintf(topic,MAX_MQTT_LEN,"%s%s", MQTT_SUBSCRIBE_TOPIC_PREFIX,clientId);
+						MqttCMInfo("The value of topic is %s\n",topic);
+					}
 					MqttCMInfo("mqtt subscribe %s\n", subscribe);
-					mqtt_subscribe();
+					if(topic !=NULL)
+						mqtt_subscribe(topic);
 					MqttCMDebug("mqtt_subscribe done\n");
 				}
+				else if(strcmp (data, "mesh") == 0 || strcmp (data, "Telemetry"))
```

Comment:
Remove other component checks as there is no use case currently , instead add a comment in code to follow same for new components